iroha-cli/interactive/impl/interactive_transaction_cli.cpp

```diff
@@ -0,0 +1,443 @@
+
+    InteractiveTransactionCli::InteractiveTransactionCli(
+        std::string creator_account) {
+      creator_ = creator_account;
+      // TODO: synchronize counter with iroha
+      tx_counter_ = 0;
+      assign_command_handlers();
+      assign_result_handlers();
+    }
+
+    void InteractiveTransactionCli::run() {
+      std::string line;
+      bool is_parsing = true;
+      current_context_ = MAIN;
+      printMenu("Forming a new transactions, choose command to add: ",
+                commands_points_);
+      // Creating a new transaction, increment local tx_counter
+      ++tx_counter_;
+      while (is_parsing) {
+        line = promtString("> ");
+        switch (current_context_) {
+          case MAIN:
+            is_parsing = parseCommand(line);
+            break;
+          case RESULT:
+            is_parsing = not parseResult(line);
+            break;
+        }
+      }
+    }
+
+    bool InteractiveTransactionCli::parseCommand(std::string line) {
+      std::transform(line.begin(), line.end(), line.begin(), ::tolower);
+      // Find in main handler map
+      auto command = parser::split(line)[0];
+      if (command == "b" || command == "0") {
+        // Return back to main menu
+        // Stop parsing
+        return false;
+      }
+      auto it = command_handlers_.find(command);
+      if (it != command_handlers_.end()) {
+        auto res = (this->*it->second)(line);
+        if (res) {
+          commands_.push_back(res);
+          current_context_ = RESULT;
+          printMenu("Command is formed. Choose what to do:", result_points_);
+        }
+      } else {
+        std::cout << "Command not found" << std::endl;
+      }
+      // Continue parsing
+      return true;
+    }
+
+    std::shared_ptr<iroha::model::Command>
+    InteractiveTransactionCli::parseAddAssetQuantity(std::string line) {
+      std::vector<std::string> notes = {"Account Id to add assets", "Asset id",
+                                        "Amount to add (integer part)",
+                                        "Amount to add (fractional part)"};
+
+      auto params = parseParams(line, ADD_ASSET_QTY, notes);
+      if (not params.has_value()) {
+        return nullptr;
+      }
+      auto account_id = params.value()[0];
+      auto asset_id = params.value()[1];
+      iroha::Amount amount;
+      auto val_int = parser::toUint64(params.value()[2]);
+      auto val_frac = parser::toUint64(params.value()[3]);
+      if (not val_int.has_value() || not val_frac.has_value()) {
+        std::cout << "Wrong format for amount" << std::endl;
+        return nullptr;
+      }
+      amount.int_part = val_int.value();
+      amount.frac_part = val_frac.value();
+      return generator_.generateAddAssetQuantity(account_id, asset_id, amount);
+    }
+
+    std::shared_ptr<iroha::model::Command>
+    InteractiveTransactionCli::parseAddPeer(std::string line) {
+      std::vector<std::string> notes = {"Full address of a peer", "Public key"};
```

Comment:
Instantiation of notes, params and checking value, looks boiler plate, change it with own method.